### PR TITLE
find-host command

### PR DESCRIPTION
**Pull request by pke11y**
This PR adds a new command to find a host in the inventory table. The use case was outlined by IP Fabric.

Current filters are `ip` and `mac` validated using `netutils`.

### CODE REVIEW

File: nautobot_chatops_ipfabric/ipfabric.py [hunk 1]
```diff
@@ -324,3 +324,46 @@ def get_wireless_ssids(self, snapshot_id="$last", limit=DEFAULT_PAGE_LIMIT):
         }
 
         return self.get_response("/api/v1/tables/wireless/radio", payload)
+
+    def get_host(self, search_key, search_value, snapshot_id="$last", limit=DEFAULT_PAGE_LIMIT):
```
Comment (chadell) [inline nautobot_chatops_ipfabric/ipfabric.py:328]: not for now, but maybe this (`DEFAULT_PAGE_LIMIT`) is a good candidate for Plugin settings

File: nautobot_chatops_ipfabric/ipfabric.py [hunk 2]
```diff
@@ -324,3 +324,46 @@ def get_wireless_ssids(self, snapshot_id="$last", limit=DEFAULT_PAGE_LIMIT):
...
+        """Return inventory host information."""
+        logger.debug("Received host inventory request")
```
Comment (chadell) [inline nautobot_chatops_ipfabric/ipfabric.py:330]: maybe we could also log the arguments? (search_key and search_value)

File: nautobot_chatops_ipfabric/ipfabric.py [hunk 3]
```diff
@@ -324,3 +324,46 @@ def get_wireless_ssids(self, snapshot_id="$last", limit=DEFAULT_PAGE_LIMIT):
...
+            "filters": {search_key: ["eq", search_value]},
+            "pagination": {"limit": limit, "start": 0},
+            "snapshot": snapshot_id,
+        }
+        logger.debug("Requesting host inventory with payload: %s", payload)
+        return self.get_response("/api/v1/tables/addressing/hosts", payload)
+
+    def find_host(self, search_key, search_value, snapshot_id="$last", limit=DEFAULT_PAGE_LIMIT):
+        """Get and parse inventory host information."""
+        logger.debug("Parsing host inventory request")
```
Comment (chadell) [inline nautobot_chatops_ipfabric/ipfabric.py:355]: smae log comment as above

File: nautobot_chatops_ipfabric/ipfabric.py [hunk 4]
```diff
@@ -324,3 +324,46 @@ def get_wireless_ssids(self, snapshot_id="$last", limit=DEFAULT_PAGE_LIMIT):
...
+
+        hosts = self.get_host(search_key, search_value, snapshot_id, limit)
+        logger.debug("Parsing hosts: %s", hosts)
+        parsed_hosts = []
+
+        for host in hosts:
+            for edge in host.get("edges"):
+                edge_details = " ".join([edge.get("hostname", ""), edge.get("intName", "")])
+                host["edges"] = edge_details
```
Comment (chadell) [inline nautobot_chatops_ipfabric/ipfabric.py:364]: I'm a bit confused by these `fors`. It looks like we can get multiple edges and gateways, but we only store the last?? 🤔 
I guess I'm missing something here

File: nautobot_chatops_ipfabric/worker.py [hunk 5]
```diff
@@ -705,3 +717,61 @@ def get_wireless_clients(dispatcher, ssid=None, snapshot_id=None):
...
+    if not filter_value:
+        dispatcher.prompt_for_text(
+            f"{BASE_CMD} {sub_cmd} {filter_key}",
+            f"Enter a specific {filter_key} to filter by:",
+            f"{filter_key.upper()}",
+        )
+        return False
+
+    is_valid_input = inventory_host_func_mapper.get(filter_key)
+    if not is_valid_input(filter_value):
```
Comment (chadell) [inline nautobot_chatops_ipfabric/worker.py:740]: personal suggestion for better readability (at first glance, I thought `is_valid_input` as a boolean):
```suggestion
    is_valid_input_func = inventory_host_func_mapper.get(filter_key)
    if not is_valid_input_func(filter_value):
```

**Review by chadell — CHANGES_REQUESTED** [review]
my only concern is on the "fors" question, the rest looks ok

File: nautobot_chatops_ipfabric/ipfabric.py (hunk 1 above)
Comment (pke11y) [reply nautobot_chatops_ipfabric/ipfabric.py:328]: yes, good idea

File: nautobot_chatops_ipfabric/ipfabric.py (hunk 2 above)
Comment (pke11y) [reply nautobot_chatops_ipfabric/ipfabric.py:330]: will update this

File: nautobot_chatops_ipfabric/ipfabric.py (hunk 4 above)
Comment (pke11y) [reply nautobot_chatops_ipfabric/ipfabric.py:364]: For all the data I have only one object is ever returned for these attributes. Will update to account for possible other elements.

File: nautobot_chatops_ipfabric/ipfabric.py (hunk 1 above)
Comment (pke11y) [reply nautobot_chatops_ipfabric/ipfabric.py:328]: #62

**Review by chadell — APPROVED** [review]
LGTM

**Pull request merged**